randovania/game_description/data_writer.py

```diff
@@ -256,7 +259,7 @@ def write_node(node: Node) -> dict:
     :return:
     """
 
-    data = {}
+    data: dict = {}
```

Comment:
Why is the type here necessary lol. Does mypy  really not see `{}` as a dict otherwise?

Reply:
It asks to specify the key/value types of the dict, which I'm then explicitly being generic. Could probably used `dict[str, Any]` instead. Or some `JsonObject` defintiion.